### ISSUE TITLE
Upgrade from 2.2.4 to 2.3.* = breaking change?

### ISSUE THREAD

**Issue by MoonRaccoon** [opentok/Opentok-Ruby-SDK#109]
We were manually updating our Gemfile and found that upgrading the gem version from 2.2.4 to either 2.3.3 or 2.3.0 causes the app to raise an exception:

This is when using the OpenTok ruby sdk from the rails console, it looks like the relevant in line opentok/client.rb was changed in 2.3.0.

    2.2.2 :003 > session = OPENTOK.create_session
    OpenTok::OpenTokError: Failed to connect to OpenTok. Response code: uninitialized constant                          OpenTok::Client::HashExtensions
    from /Users/Blah/.rvm/gems/ruby-2.2.2/gems/opentok-2.3.3/lib/opentok/client.rb:38:in `rescue in create_session'
    from /Users/Blah/.rvm/gems/ruby-2.2.2/gems/opentok-2.3.3/lib/opentok/client.rb:27:in `create_session'
    from /Users/Blah/.rvm/gems/ruby-2.2.2/gems/opentok-2.3.3/lib/opentok/opentok.rb:166:in `create_session'

**Comment by aoberoi** [comment]
@MoonRaccoon it seems like the actual issue might be that the HashExtensions are not loaded when using the rails console. Can see if the actual rails server application is able to create sessions without this error?

**Comment by MoonRaccoon** [comment]
Sorry for the extremely late response, will test this and get back to you.

**Comment by LuckDragon82** [comment]
This seems to be caused by the rails autoloader. If you change the file structure to match the module  name of OpenTok::HashExtensions I think this will work fine (at least that what the problem seems to be).

**Comment by aoberoi** [comment]
@LuckDragon82 were you able to reproduce this? i was not, so i hadn't put in any effort to solving it.

**Comment by LuckDragon82** [comment]
I made a stand alone test rails project and I was not able to reproduce it. Inside my real project I get the following error 
```
OpenTok::OpenTokError - Failed to connect to OpenTok. Response code: uninitialized constant OpenTok::Client::HashExtensions:
  opentok (2.3.3) lib/opentok/client.rb:63:in `rescue in start_archive'
  opentok (2.3.3) lib/opentok/client.rb:42:in `start_archive'
  opentok (2.3.3) lib/opentok/archives.rb:71:in `create'
```
Clearly its something in my project, but I have no idea what I could be changing that would affect the way your gem loads. I don't know why its searching for `OpenTok::Client::HashExtensions` instead of `OpenTok::HashExtensions`

**Comment by aoberoi** [comment]
i did a little more research into the problem. i have one unverified idea about how we could fix this.

see: https://github.com/opentok/Opentok-Ruby-SDK/blob/master/opentok.gemspec#L2-L3

when this gem is loaded, it modifies the Ruby `$LOAD_PATH`. this was considered an decent practice in the Ruby 1.8.x days. noting that we have already dropped support and state minimum compatibility at 1.9.3, this is no longer the best way to load the appropriate code, and its now considered an antipattern.

here are two helpful sources of information:
*  https://practicingruby.com/articles/ways-to-load-code
*  http://guides.rubygems.org/patterns/#loading-code

@LuckDragon82 if you were willing to try to get rid of the lines that modify the `$LOAD_PATH`, and get the tests to pass, i'd be happy to merge a PR. let me know if this is something you want to try.

**Comment by LuckDragon82** [comment]
I was able to get the test to pass locally, but it doesn't seem to be requiring the files the same on your CI. Is there something I should know about the way it runs test?

**Comment by aoberoi** [comment]
Thanks @LuckDragon82! I don't think the CI issues are your fault, I've seen them for a little while now. I'll run the tests locally on each of the failing ruby versions and if they pass, the CI shouldn't be a blocker. 

Then I just need to figure out what's wrong with Travis... :/

**Comment by LuckDragon82** [comment]
I got its down 3/5 passing I don't think ruby 1.9.3 likes require relative. No idea whats wrong with 2.2.2

**Comment by LuckDragon82** [comment]
sorry for all the changes... Never messed around with all the different ways to require, and I don't see the same issues locally so I need to push it to CI to see what happens.

**Comment by aoberoi** [comment]
@LuckDragon82: doesn't bother me at all, i'm happy you're being thorough!

**Comment by aoberoi** [comment]
@LuckDragon82: i think we are actually running into a bug with Travis. see #114. the PR that has a fix is building, but if it works and i merge to master, try updating your branch with the latest master and we might resolve this issue.

**Comment by aoberoi** [comment]
fixed by #113

**Comment by aoberoi** [comment]
fix is in v2.3.4
